src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -148,7 +171,26 @@ Status MediapipeGraphDefinition::create(std::shared_ptr<MediapipeGraphExecutor>&
         return status;
     }
     SPDLOG_DEBUG("Creating Mediapipe graph executor: {}", getName());
-    pipeline = std::make_shared<MediapipeGraphExecutor>(getName(), std::to_string(getVersion()), this->config);
+
+    bool passKfsRequestFlag = false;
+    // Check if we are passing whole KFS request and response
+    if (this->config.input_stream().size() > 0) {
+        std::string firstName = this->config.input_stream()[0];
+        if (startsWith(firstName.c_str(), "REQUEST:")) {
+            if (this->config.output_stream().size() > 0) {
+                firstName = this->config.output_stream()[0];
+                if (startsWith(firstName.c_str(), "RESPONSE:")) {
+                    SPDLOG_DEBUG("KServe for mediapipe graph passing whole KFS request graph detected.");
+                    passKfsRequestFlag = true;
+                } else {
+                    SPDLOG_DEBUG("KServe for mediapipe graph passing whole KFS request and response requires RESPONSE: string in the output stream name");
+                    return Status(StatusCode::MEDIAPIPE_KFS_PASS_MISSING_RESPONSE_GRAPH_OUTPUT_NAME);
+                }
+            }
+        }
+    }
```

Comment:
You could encapsulate whole logic inside helper function Status setKFSPassthrough(bool& passKfsRequestFlag, config.input_stream(), config.output_stream()), and then test it at lower level.

Right now is it possible to set this flag to true if we have 2 inputs streams & output streams
eg.
input_stream:REQUEST:in1
input_stream:IMAGE:in2
output_stream:RESPONSE:out1
output_stream:IMAGE:out2
?

Reply:
Also MEDIAPIPE_KFS_PASS_MISSING_RESPONSE_GRAPH_OUTPUT_NAME is misleading here. Maybe:
MEDIAPIPE_KFS_PASSTHROUGH_MISSING_OUTPUT_RESPONSE_TAG?

Reply:
Added check for 1 input and 1 output name in streams and created  a function.